### PR TITLE
Feature/4132 account details full screen

### PR DESCRIPTION
**Pull request by ektaghag-eaton**
Fixes 4132 .

#### Changes proposed in this Pull Request:

- Account Detail screen
-
-

#### Screenshots / Screen Recording (if applicable)

-

#### To Test:

- http://localhost:3000/registration-workflow

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Comment by manojleaton** [comment]
Validation error messages are not translated

Screenshot 2023-06-26 at 1 07 48 PM

**Comment by ektaghag-eaton** [comment]
> Validation error messages are not translated
> 
> Screenshot 2023-06-26 at 1 07 48 PM

Fixed error please check

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    const [firstName, setFirstName] = useState(screenData.AccountDetails.firstName ?? '');
+    const [lastName, setLastName] = useState(screenData.AccountDetails.lastName ?? '');
+
+    const {
+        title = t('bluiRegistration:REGISTRATION.STEPS.ACCOUNT_DETAILS'),
+        instructions = t('bluiRegistration:REGISTRATION.INSTRUCTIONS.ACCOUNT_DETAILS'),
+        firstNameLabel = t('bluiCommon:FORMS.FIRST_NAME'),
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+        initialFirstName,
+        initialLastName,
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:24]: You can remove these prop variables from here right? Anyway you are using these props below in the code.

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            WorkflowCardHeaderProps={{ title: title }}
+            WorkflowCardInstructionProps={{ instructions: instructions }}
+            initialFirstName={firstName}
+            initialLastName={lastName}
+            firstNameLabel={firstNameLabel}
+            firstNameValidator={firstNameValidator}
+            lastNameLabel={lastNameLabel}
+            lastNameValidator={lastNameValidator}
+            firstNameTextFieldProps={firstNameTextFieldProps}
+            lastNameTextFieldProps={lastNameTextFieldProps}
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:82]: Passing `firstNameTextFieldProps` and `lastNameTextFieldProps` props in here is not needed as you are not passing any specific prop inside of it.

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    const {
+        title = t('bluiRegistration:REGISTRATION.STEPS.ACCOUNT_DETAILS'),
+        instructions = t('bluiRegistration:REGISTRATION.INSTRUCTIONS.ACCOUNT_DETAILS'),
+        firstNameLabel = t('bluiCommon:FORMS.FIRST_NAME'),
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+        initialFirstName,
+        initialLastName,
+        lastNameLabel = t('bluiCommon:FORMS.LAST_NAME'),
+        firstNameTextFieldProps,
+        lastNameTextFieldProps,
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:27]: You can remove these props `firstNameTextFieldProps` and `lastNameTextFieldProps`, as you are not passing any specific prop inside of it.

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            return t('bluiCommon:FORMS.FIRST_NAME_LENGTH_ERROR');
+        },
+        lastNameValidator = (name: string): boolean | string => {
+            if (name?.length > 2) {
+                setLastName(name);
+                return true;
+            }
+            return t('bluiCommon:FORMS.LAST_NAME_LENGTH_ERROR');
+        },
+        sx,
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:42]: You can remove `sx` prop from here as you are not passing any styles inside of it

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            WorkflowCardInstructionProps={{ instructions: instructions }}
+            initialFirstName={firstName}
+            initialLastName={lastName}
+            firstNameLabel={firstNameLabel}
+            firstNameValidator={firstNameValidator}
+            lastNameLabel={lastNameLabel}
+            lastNameValidator={lastNameValidator}
+            firstNameTextFieldProps={firstNameTextFieldProps}
+            lastNameTextFieldProps={lastNameTextFieldProps}
+            sx={sx}
```
Comment (manojleaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:83]: Passing `sx` prop is not needed in here as you are not adding any styles inside of it

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+type AccountDetailsFullScreenProps = AccountDetailsScreenProps & {
+    title?: string;
+    instructions?: string;
+};
+
+export const AccountDetailsScreen: React.FC<AccountDetailsFullScreenProps> = (props) => {
+    const { t } = useLanguageLocale();
+    const regWorkflow = useRegistrationWorkflowContext();
+    const { nextScreen, previousScreen, screenData } = regWorkflow;
+    const [firstName, setFirstName] = useState(screenData.AccountDetails.firstName ?? '');
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:15]: Please destructure to:
`const {firstName, lastName} = screenData.AccountDetails;`

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            nextScreen({
+                screenId: 'AccountDetails',
+                values: { firstName: firstName, lastName: lastName },
+            });
+        } catch {
+            console.error('Error while updating account details...');
+        }
+    };
+
+    const onPrevious = (): void => {
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:58]: There is no need of try catch block onPrevious

File: login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            lastNameValidator={lastNameValidator}
+            firstNameTextFieldProps={firstNameTextFieldProps}
+            lastNameTextFieldProps={lastNameTextFieldProps}
+            sx={sx}
+            WorkflowCardActionsProps={{
+                canGoPrevious: true,
+                showPrevious: true,
+                previousLabel: t('bluiCommon:ACTIONS.BACK'),
+                onPrevious: onPrevious,
+                canGoNext: true,
```
Comment (surajeaton) [inline login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreen.tsx:89]: We need not pass canGoNext as true, as we are passing the validator function, on base screen it should validate.

**Review by surajeaton — COMMENTED** [review]
There are few points about account details base screen:
There should not be any default values set in base screen, everything should be passed as props.
example:
- `firstNameValidator` and `lastNameValidator` should not have default set in base screen.
- There should not be `sx` defined in the props. If we want to pass `sx`, that should go in the individual workflow components like: `WorkflowCardBaseProps: { sx = { mt: 1 } },`

I know, it is out of the scope of this ticket, but anyhow we need to fix it in the same PR.

**Comment by ektaghag-eaton** [comment]
Hi @surajeaton and @manojleaton I have fixed the PR comments please review. Thanks!!!

**Comment by ektaghag-eaton** [comment]
@surajeaton  I have raised an issue https://github.com/etn-ccis/blui-react-workflows/issues/330 for fixing the Next button enable/disable while switching screens within a workflow which needs to be fixed in the Base component.

**Review by surajeaton — APPROVED** [review]
Looks good.

**Pull request merged**